src/modules/enrollment/components/staffAssignment/StaffAssignmentCard.tsx

```diff
@@ -0,0 +1,113 @@
+import EditIcon from '@mui/icons-material/Edit';
+import { Divider, Stack, Typography } from '@mui/material';
+import React, { useMemo, useState } from 'react';
+import {
+  CommonDetailGridContainer,
+  CommonDetailGridItem,
+} from '@/components/elements/CommonDetailGrid';
+import Loading from '@/components/elements/Loading';
+import TitleCard from '@/components/elements/TitleCard';
+import { useEnrollmentDashboardContext } from '@/components/pages/EnrollmentDashboard';
+import IconButtonContainer from '@/modules/enrollment/components/IconButtonContainer';
+import EditStaffAssignmentDialog from '@/modules/enrollment/components/staffAssignment/EditStaffAssignmentDialog';
+import {
+  RelationshipToHoH,
+  StaffAssignmentDetailsFragment,
+  useGetHouseholdStaffAssignmentsQuery,
+} from '@/types/gqlTypes';
+
+interface StaffAssignmentCardProps {
+  householdId: string;
+}
+
+const StaffAssignmentCard: React.FC<StaffAssignmentCardProps> = ({
+  householdId,
+}) => {
+  const { enrollment } = useEnrollmentDashboardContext();
+
+  const {
+    data: { household } = {},
+    loading,
+    error,
+  } = useGetHouseholdStaffAssignmentsQuery({
+    variables: { id: householdId },
+  }); // fixme - this query gets repeated twice with the same variables; why?
```

Comment:
@ttoomey when you review this, could you take a look at this and let me know if you have any ideas to improve it? I'm flummoxed - even memoizing the `StaffAssignmentCard` component on the `householdId` does not seem to fix it.

Network tab:
3rd query includes `GetHouseholdStaffAssignments`
![Screenshot 2024-07-26 at 8 43 36 AM](https://github.com/user-attachments/assets/643fc95e-953b-47e8-8531-9e33449d9000)

4th query asks for the same data, even though the component is already loaded and visible:
![Screenshot 2024-07-26 at 8 44 20 AM](https://github.com/user-attachments/assets/19bbaf99-bd08-406f-bcaf-bb76cf7fd269)

Reply:
@martha Thanks for finding this repeated query!

It was indeed a bit tricky to track this down. So I believe what's happening is that another query, `GetHouseholdClientAlerts`, is running on the same page; this query seems to be missing the id field on the household. This is important because Apollo's query cache uses the id (or configured keyFields) to determine how to cache the results.

When this second query resolves, it seems like the behavior is that since Apollo can't determine how to cache the query results, it reruns the initial household query. I'm not sure if this is a bug or intentional behavior in Apollo itself, maybe it's trying to be safe and just invalidating the household cache.

Generally all queries should resolve the id or key fields. Usually this is part of the fragment already but in this case they need to be included explicitly.

This seems to fix the issue for me locally:
```diff
diff --git a/src/api/operations/alert.queries.graphql b/src/api/operations/alert.queries.graphql
index 19462745..4e2e805f 100644
--- a/src/api/operations/alert.queries.graphql
+++ b/src/api/operations/alert.queries.graphql
@@ -1,6 +1,8 @@
 query GetHouseholdClientAlerts($id: ID!) {
   household(id: $id) {
+    id
     householdClients {
+      id
       client {
         ...ClientWithAlertFields
       }
diff --git a/src/types/gqlTypes.ts b/src/types/gqlTypes.ts
index b10c2521..32e3db47 100644
--- a/src/types/gqlTypes.ts
+++ b/src/types/gqlTypes.ts
@@ -8189,8 +8189,10 @@ export type GetHouseholdClientAlertsQuery = {
   __typename?: 'Query';
   household?: {
     __typename?: 'Household';
+    id: string;
     householdClients: Array<{
       __typename?: 'HouseholdClient';
+      id: string;
       client: {
         __typename?: 'Client';
         id: string;
@@ -35089,7 +35091,9 @@ export type DeleteClientAlertMutationOptions = Apollo.BaseMutationOptions<
 export const GetHouseholdClientAlertsDocument = gql`
   query GetHouseholdClientAlerts($id: ID!) {
     household(id: $id) {
+      id
       householdClients {
+        id
         client {
           ...ClientWithAlertFields
         }
```

Reply:
Stellar, thank you so much @ttoomey for the sleuthing and for the thorough explanation too!